lib/openhab/core/dsl/monkey_patch/types/decimal_type.rb

```diff
@@ -10,15 +10,51 @@ class Java::OrgOpenhabCoreLibraryTypes::DecimalType
   # rubocop:enable Style/ClassAndModuleChildren
 
   #
-  # Compare self to other using Java BigDecimal compare method
+  # Compare DecimalType to supplied object
   #
   # @param [Object] other object to compare to
   #
-  # @return [Boolean] True if have the same BigDecimal representation, false otherwise
+  # @return [Integer] -1,0,1 or nil depending on value supplied, nil comparison to supplied object is not possible.
   #
-  def ==(other)
-    return equals(other) unless other.is_a? Integer
+  def <=>(other)
+    logger.trace("#{self.class} #{self} <=> #{other} (#{other.class})")
+    case other
+    when Integer, Float
+      to_big_decimal.compare_to(Java::JavaMath::BigDecimal.new(other))
+    when Java::OrgOpenhabCoreTypes::UnDefType
+      1
```

Comment:
I don't know the answer here.. is 1 right right answer or is nil here?

Reply:
If it returns nil, the equality comparison will raise an error because nil class doesn't have <=> operator. I picked 1 so anything is greater than undef/null.